BL-383 - fix edit tabs - only show if permission and return url

randyhoulahan opened the pull request.

randyhoulahan committed:
BL-383 - fix edit tabs - only show if permission and return url

sbilodeau commented on components/page/body/tabs.vue:
```diff
@@ -48,13 +46,32 @@
 
+    const baseUrl   = computed(()=>siteStore.localizedHost+getUrlComponent());
+
+    const showSelf = computed(() => pageStore?.isSystemPage? meStore?.showEditSystemPages : meStore?.showEdit)
```
too many `?` 😆

randyhoulahan replied:
fixed https://github.com/scbd/bioland-head/commit/385feaa

sbilodeau commented on components/page/body/tabs.vue:
```diff
@@ -48,13 +46,32 @@
+
+    const isContributorCanEdit = computed(() => isContributor.value && pageStore?.page?.uid?.meta?.drupal_internal__target_id === meStore?.diuid && !pageStore?.page?.status);
+
+    const cloneUrl = computed(()=>siteStore.localizedHost+`/clone/${pageStore?.page?.drupalInternalNid}/quick_clone`+returnUrl.value);
```
what returnUrl value contains?
`?returnUrl=somethings..?`. the name is not reflecting the value thosu shoudl be only an url

randyhoulahan replied:
fixed https://github.com/scbd/bioland-head/commit/385feaa

randyhoulahan committed:
Merge branch 'master' into BL-383-

randyhoulahan committed:
BL-383 - refactor tabs.vue to address review comments

- Reduce excessive optional chaining operators (?.) per sbilodeau feedback
- Rename returnUrl to contain only the encoded path instead of full query param
- Remove duplicate baseUrl and showSelf declarations
- Fix inconsistent spacing around && operator
- Add encodeURIComponent for URL segments for safety
